snapcraft/tests/test_lifecycle.py

```diff
@@ -689,7 +727,7 @@ def test_prime_with_build_info_records_snapcraft_yaml(self):
 build-packages: []
 """.format(self.project_options.deb_arch))
         self.assertThat(
-            os.path.join('prime', 'snap', 'snapcraft.yaml'),
+            os.path.join('prime', 'snap', 'manifest.yaml'),
```

Comment:
It's the other way around. manifest will include more information.
But that's what all the tests are checking. I added here one test to check that the recorded snapcraft.yaml is a pristine copy, and all the tests check pieces of information that are added to the original and then recorded in manifest.yaml